### ISSUE TITLE
Make it clear that Astro Actions (no JS only) are limited by the size of the browsers cookies

### ISSUE THREAD

**Issue by rstainsby** [withastro/docs#9766]
### 📚 Subject area/topic

Actions

### 📋 Page(s) affected (or suggested, for new content)

https://docs.astro.build/en/guides/actions/

### 📋 Description of content that is out-of-date or incorrect

Astro actions, when using form actions i.e. no JavaScript, appear to use cookies to store the payload returned by the action.

This works but cookies have a size limit that is set by the browser of ~4kb. This means that if you have a large response (in my case an LLM transcription) the request is successful but no data is available on the client, with no obvious reason why.

This is likely a niche scenario but it's taken me hours to debug, can it please be made clear that actions use cookies and that they shouldn't be used for large responses?

**Comment by sarah11918** [comment]
Hi, thank you for reporting this! I'll tag @bholmesdev in here for comment because docs always documents expected, intended behavior and maybe this isn't intended, or maybe there's an alternative we should be documenting, or maybe this means we need some error handling in the code... or maybe there's some work around entirely at the code level.

Ben, what `action` would you suggest docs take here?

**Comment by bholmesdev** [comment]
Thanks for flagging! We do have a warning in the docs that cookies are used, but we should definitely call out the size limit as well. Still, I think an error message in development would be _far_ more helpful than a docs note you have to find. I think we can check the serialized payload size to catch these cases early. I'll make a note for core about this.

**Comment by rstainsby** [comment]
Hi @bholmesdev. I actually missed the warning about using cookies in the docs the first time I looked at it. So the error message in development definitely feels like the best solution! If you're happy for me to take a look at it, I can make a PR to implement this?

I've made the changes to the docs and submitted [this PR](https://github.com/withastro/docs/pull/9781). @sarah11918 let me know if there's anything I've missed.
